helix-term/src/commands.rs

```diff
@@ -5371,3 +5380,181 @@ fn replay_macro(cx: &mut Context) {
         cx.editor.macro_replaying.pop();
     }));
 }
+
+fn jump_to_identifier_label(cx: &mut Context) {
+    let jump_targets = find_all_identifiers_in_view(cx);
+    jump_with_targets(cx, jump_targets, false);
+}
+
+fn jump_to_identifier_label_and_extend_selection(cx: &mut Context) {
+    let jump_targets = find_all_identifiers_in_view(cx);
+    jump_with_targets(cx, jump_targets, true);
+}
+
+fn jump_to_str_label(cx: &mut Context) {
+    find_and_jump_to_str(cx, false);
+}
+
+fn jump_to_str_label_and_extend_selection(cx: &mut Context) {
+    find_and_jump_to_str(cx, true);
+}
+
+fn find_and_jump_to_str(cx: &mut Context, extend_selection: bool) {
+    jump::setup(cx);
+    let nchars = doc!(cx.editor)
+        .config
+        .load()
+        .jump_mode
+        .num_chars_before_label as usize;
+    cx.editor.set_status(format!("Press {} key(s)", nchars));
+    cx.on_next_key(move |cx, event| {
+        jump_with_str_input(
+            cx,
+            event,
+            extend_selection,
+            String::with_capacity(nchars),
+            nchars,
+        )
+    });
+}
+
+fn jump_with_str_input(
+    cx: &mut Context,
+    event: KeyEvent,
+    extend_selection: bool,
+    mut s: String,
+    nchars: usize,
+) {
+    let c = event.char().filter(|c| c.is_ascii());
+    if c.is_none() {
+        cleanup(cx);
+        return;
+    }
+    let c = c.unwrap();
```

Comment:
```suggestion
    let Some(c) = event.char().filter(|c| c.is_ascii()) else {
        cleanup(cx);
        return
    };
```

You can use `let-else` here to simplify this a lot.